lib/active_hash/conditions.rb

```diff
@@ -0,0 +1,21 @@
+class ActiveHash::Relation::Conditions
+  attr_reader :conditions
+
+  delegate_missing_to :conditions
```

Comment:
Sure. This class is an delegator to an array of conditions. It delegates methods to the array of conditions, but also implements `#matches?` which is a short-hand for `array_of_conditions.all? { |c| c.matches?(record) }` and makes it a bit cleaner to work with in the `Relation`.

It also adds `.wrap` which is similar to [`Array.wrap`](https://apidock.com/rails/v5.2.3/Array/wrap/class) to 'wrap' an array of conditions (or return itself if already wrapped).

Happy to add comments to the code, if you believe this would make it clearer.